### ISSUE TITLE
alamofire for self-signed certificates

### ISSUE THREAD

**Issue by psankar** [Alamofire/Alamofire#2402]
## What did you do?

Connect to a self-signed HTTPS server running on localhost

## What did you expect to happen?

Connection to happen succesfully

## What happened instead?

I get a `ATS failed system trust` and related errors. 

My sources:
```
let Almgr : Alamofire.SessionManager = {
    // Create the server trust policies
    let serverTrustPolicies: [String: ServerTrustPolicy] = [
        "localhost": .disableEvaluation
    ]
    // Create custom manager
    let configuration = URLSessionConfiguration.default
    configuration.httpAdditionalHeaders = Alamofire.SessionManager.defaultHTTPHeaders
    let man = Alamofire.SessionManager(
        configuration: URLSessionConfiguration.default,
        serverTrustPolicyManager: ServerTrustPolicyManager(policies: serverTrustPolicies)
    )
    return man
}()

 Almgr.request(url, method: .post, parameters: params).responseJSON {
            response in

            if response.result.isSuccess {
                print("Success")
            } else {
                print("Failure")
            }
        }
```

The full output is:

```
2017-12-30 18:24:20.114486+0530 myApp[58036:2721102] ATS failed system trust
2017-12-30 18:24:20.114625+0530 myApp[58036:2721102] System Trust failed for [1:0x600000178a80]
2017-12-30 18:24:20.114814+0530 myApp[58036:2721102] TIC SSL Trust Error [1:0x600000178a80]: 3:0
2017-12-30 18:24:20.115142+0530 myApp[58036:2721102] NSURLSession/NSURLConnection HTTP load failed (kCFStreamErrorDomainSSL, -9802)
2017-12-30 18:24:20.115274+0530 myApp[58036:2721102] Task <4E3D9E88-B9CE-48C4-850C-5A3E7C9A6A72>.<1> HTTP load failed (error code: -1200 [3:-9802])
2017-12-30 18:24:20.115469+0530 myApp[58036:2721231] Task <4E3D9E88-B9CE-48C4-850C-5A3E7C9A6A72>.<1> finished with error - code: -1200
```

I am opening this bug report, as a documentation issue, where "Using alamofire to connect to a self-signed HTTPS server" can come as a step. I would even be ready to contribute this as a patch once I figure out the solution.

## Alamofire Environment

**Alamofire version:** I installed with `pod 'Alamofire', '~> 4.5'`
**Xcode version:** Version 9.1 (9B55)
**Swift version:** 4.0
**Platform(s) running Alamofire:**
**macOS version running Xcode:** mac OS High Sierra Version 10.13.2 (17C88)

**Comment by psankar** [comment]
I figured out the solution. For localhost, self-signed https certificates, we need to edit the Info.plist file and add the following snippet, to whitelist localhost from SSL errors.

```
<key>NSAppTransportSecurity</key>
    <dict>
        <key>NSAllowsLocalNetworking</key>
        <true/>
    </dict>
```

If you believe that this issue is worth documenting, I will be happy to give a PR to the README. Feel free to close this otherwise. Thanks.

**Comment by jshier** [comment]
@psankar Yes, a PR would be appreciated. You can add this info to our existing info on getting the self signed certificates working. Apple documentation on that key would be helpful as well.

**Comment by jshier** [comment]
This was addressed in #2462.
